src/main/java/com/rapid7/client/dcerpc/msrrp/RegistryService.java

```diff
@@ -171,21 +200,54 @@ protected String canonicalize(String keyPath) {
         }
     }
 
-    protected byte[] openKey(final String hiveName, final String keyPath) throws IOException {
+    public byte[] openKey(final String hiveName, final String keyPath) throws IOException {
+        return openKey(hiveName, keyPath, ACCESS_MASK);
+    }
+
+    public byte[] openKey(final String hiveName, final String keyPath, EnumSet<AccessMask> access) throws IOException {
```

Comment:
I'd rather not have `EnumSet<AccessMask>` in our service but if other registry elements are doing it we can refactor it out later.

Reply:
Fixed